src/app/analytics/analytics.factory.js

```diff
@@ -448,12 +445,12 @@ const analyticsFactory = /* @ngInject */ function ($window, $timeout, envService
       $window.digitalData.product = product
       $window.dataLayer = $window.dataLayer || []
       $window.dataLayer.push({
-        event: 'give-gift-modal',
+        event: 'view_item',
         ecommerce: {
           currencyCode: 'USD',
-          detail: {
-            products: [generateProduct(modifiedProductData)]
-          }
+          // value is unavailable until the user selects a gift amount
```

Comment:
Will there ever be a `view_item` which has a value? If no, should we have it this object, or do the analytics guys need it?

Reply:
Good question. `view_item` won't ever have a `value` amount at this stage because the user hasn't selected the gift amount or frequency yet. When I implemented the GA4 spec for branded checkout, the spec docs said, "If no value exists for an attribute in the dataLayer, please pass JavaScript `undefined` as the value." That's why I'm passing `undefined` instead of omitting that field.